brave/src/main/java/brave/internal/propagation/CorrelationFieldScopeDecorator.java

```diff
@@ -13,85 +13,176 @@
  */
 package brave.internal.propagation;
 
+import brave.Tracer;
+import brave.internal.InternalPropagation;
 import brave.internal.Nullable;
+import brave.internal.PropagationFields;
+import brave.propagation.CurrentTraceContext;
 import brave.propagation.CurrentTraceContext.Scope;
 import brave.propagation.CurrentTraceContext.ScopeDecorator;
+import brave.propagation.ExtraFieldPropagation;
 import brave.propagation.TraceContext;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Set;
 
 /**
- * Adds correlation properties "traceId", "parentId", "spanId" and "sampled" when a {@link
- * brave.Tracer#currentSpan() span is current}. "traceId" and "spanId" are used in log correlation.
+ * Adds correlation properties "traceId", "parentId", "spanId" and "sampled" when a {@linkplain
+ * Tracer#currentSpan() span is current}. "traceId" and "spanId" are used in log correlation.
  * "parentId" is used for scenarios such as log parsing that reconstructs the trace tree. "sampled"
  * is used as a hint that a span found in logs might be in Zipkin.
  */
 public abstract class CorrelationFieldScopeDecorator implements ScopeDecorator {
+  final Updater[] updaters;
+
+  protected static abstract class Builder<B extends Builder<B>> {
+    final Set<String> extraFields = new LinkedHashSet<>(); // insertion order;
+
+    /**
+     * Adds a field from {@link ExtraFieldPropagation} into the correlation context.
```

Comment:
Just want to confirm `ExtraFieldPropagation` is the only type of propagation this would ever make sense for?

Reply:
it might not be the only possible thing but it is what's often requested leading to the sleuth thing a couple years ago.

Reply:
after thinking about this, yeah I believe it is better to be explicit as we have nothing else and it is hard to predict when we could.